diff --git a/dist/types.d.ts b/dist/types.d.ts
index e54284755aa19678d6176b510984190c879d6cee..253a0480f347989afdf0a56313299c664cf5f551 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -1,10 +1,5 @@
 import type { ModuleHooks, ModulePublicRuntimeConfig } from './module'
 
-declare module '@nuxt/schema' {
-  interface NuxtHooks extends ModuleHooks {}
-  interface PublicRuntimeConfig extends ModulePublicRuntimeConfig {}
-}
-
 declare module 'nuxt/schema' {
   interface NuxtHooks extends ModuleHooks {}
   interface PublicRuntimeConfig extends ModulePublicRuntimeConfig {}
