diff --git a/dist/runtime/server/routes/_ipx.js b/dist/runtime/server/routes/_ipx.js
--- a/dist/runtime/server/routes/_ipx.js
+++ b/dist/runtime/server/routes/_ipx.js
@@ -1,8 +1,8 @@
 import { fileURLToPath } from "node:url";
 import { createIPX, createIPXH3Handler, ipxFSStorage, ipxHttpStorage } from "ipx";
-import { lazyEventHandler, useBase } from "h3";
+import { lazyEventHandler, withBase } from "h3";
 import { isAbsolute } from "pathe";
 import { useRuntimeConfig } from "#imports";
 export default lazyEventHandler(() => {
   const opts = useRuntimeConfig().ipx || {};
   const fsDir = opts?.fs?.dir ? (Array.isArray(opts.fs.dir) ? opts.fs.dir : [opts.fs.dir]).map((dir) => isAbsolute(dir) ? dir : fileURLToPath(new URL(dir, import.meta.url))) : void 0;
@@ -18,5 +18,5 @@ export default lazyEventHandler(() => {
   };
   const ipx = createIPX(ipxOptions);
   const ipxHandler = createIPXH3Handler(ipx);
-  return useBase(opts.baseURL, ipxHandler);
+  return withBase(opts.baseURL, ipxHandler);
 });
