diff --git a/dist/index.mjs b/dist/index.mjs
index c016fbace9a6b0bc451e96439934a71007f84ec2..74f5816602aec60d6f504577bae5e0c39d356530 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -665,7 +665,7 @@ async function bundle(nuxt) {
 		base: nitroConfig.serverDir,
 		watchOptions: { ignored: [isIgnored] }
 	};
-	const cacheDriverPath = join(distDir, "runtime/utils/cache-driver.js");
+	const cacheDriverPath = join(distDir, "runtime/utils/cache-driver.mjs");
 	const cacheDriverOption = isWindows ? pathToFileURL(cacheDriverPath).href : cacheDriverPath;
 	if (nuxt.options.dev) {
 		const payloadCacheDir = resolve(nuxt.options.buildDir, "cache/nuxt/payload");
