diff --git a/dist/runtime/server/api.js b/dist/runtime/server/api.js
index 3685dcb..8732457 100644
--- a/dist/runtime/server/api.js
+++ b/dist/runtime/server/api.js
@@ -1,9 +1,9 @@
 import { getIcons } from "@iconify/utils";
 import { hash } from "ohash";
 import { createError } from "h3";
+import { defineCachedHandler } from "nitro/cache";
 import { parseQuery, parsePath } from "ufo";
 import { consola } from "consola";
-import { useAppConfig, defineCachedEventHandler } from "#imports";
 import { collections } from "#nuxt-icon-server-bundle";
 const warnOnceSet = /* @__PURE__ */ new Set();
 const DEFAULT_ENDPOINT = "https://api.iconify.design";
@@ -14,8 +14,11 @@ function getInstallCommand(pkg) {
   if (ua.startsWith("bun")) return `bun add -D ${pkg}`;
   return `npm i -D ${pkg}`;
 }
-export default defineCachedEventHandler(async (event) => {
-  const options = useAppConfig().icon;
+export default defineCachedHandler(async (event) => {
+  const options = {
+    iconifyApiEndpoint: DEFAULT_ENDPOINT,
+    fallbackToApi: true
+  };
   const collectionName = event.context.params?.collection?.replace(/\.json$/, "");
   const collection = collectionName && Object.hasOwn(collections, collectionName) ? await collections[collectionName]?.() : null;
   const apiEndPoint = options.iconifyApiEndpoint || DEFAULT_ENDPOINT;
diff --git a/dist/runtime/plugin.js b/dist/runtime/plugin.js
index bc7a295..52f0c8d 100644
--- a/dist/runtime/plugin.js
+++ b/dist/runtime/plugin.js
@@ -1,10 +1,11 @@
 import { addAPIProvider, _api, setCustomIconsLoader } from "@iconify/vue";
-import { defineNuxtPlugin, useAppConfig, useRuntimeConfig } from "#imports";
+import { defineNuxtPlugin, useAppConfig, useRequestFetch, useRuntimeConfig } from "#imports";
 export default defineNuxtPlugin({
   name: "@nuxt/icon",
   setup() {
     const configs = useRuntimeConfig();
     const options = useAppConfig().icon;
+    const $fetch = useRequestFetch();
     _api.setFetch($fetch.native);
     const resources = [];
     if (options.provider === "server") {
