diff --git a/dist/runtime/server/kit.js b/dist/runtime/server/kit.js
index a33b35dec16261a799c40c1ff4718d03b9e96c99..23cc1078620a475d97b5950fa6ae4fe0d2b36fbc 100644
--- a/dist/runtime/server/kit.js
+++ b/dist/runtime/server/kit.js
@@ -1,5 +1,5 @@
 import { defu } from "defu";
-import { useRuntimeConfig } from "nitropack/runtime";
+import { useRuntimeConfig } from "nitro/runtime-config";
 import { createRouter as createRadixRouter, toRouteMatcher } from "radix3";
 import { parseURL, withoutBase, withoutTrailingSlash } from "ufo";
 function withoutQuery(path) {
diff --git a/dist/runtime/server/plugins/compression.js b/dist/runtime/server/plugins/compression.js
index 5952365cc4a618af5ac96276a5b6694d9a808283..d7e8e6df6bdbb92e148dcfbc09d8870cbc7643bf 100644
--- a/dist/runtime/server/plugins/compression.js
+++ b/dist/runtime/server/plugins/compression.js
@@ -1,5 +1,5 @@
 import { getRequestHeader, setResponseHeader } from "h3";
-import { defineNitroPlugin } from "nitropack/runtime";
+import { definePlugin } from "nitro";
 function getPreferredEncoding(event) {
   const acceptEncoding = getRequestHeader(event, "accept-encoding") || "";
   if (acceptEncoding.includes("gzip"))
@@ -8,7 +8,7 @@ function getPreferredEncoding(event) {
     return "deflate";
   return null;
 }
-export default defineNitroPlugin((nitro) => {
+export default definePlugin((nitro) => {
   nitro.hooks.hook("beforeResponse", async (event, response) => {
     if (!event.context._isSitemap || !response.body)
       return;
diff --git a/dist/runtime/server/plugins/nuxt-content-v2.js b/dist/runtime/server/plugins/nuxt-content-v2.js
index b63b55dd4d499b356a007000df9ad72cb48b720b..224da118b8a0a3691f9dfa18754a139818f6d18a 100644
--- a/dist/runtime/server/plugins/nuxt-content-v2.js
+++ b/dist/runtime/server/plugins/nuxt-content-v2.js
@@ -1,7 +1,7 @@
 import { defu } from "defu";
-import { defineNitroPlugin } from "nitropack/runtime";
+import { definePlugin } from "nitro";
 import { useSitemapRuntimeConfig } from "../utils.js";
-export default defineNitroPlugin((nitroApp) => {
+export default definePlugin((nitroApp) => {
   const { discoverImages, isNuxtContentDocumentDriven } = useSitemapRuntimeConfig();
   nitroApp.hooks.hook("content:file:afterParse", async (content) => {
     const validExtensions = ["md", "mdx"];
diff --git a/dist/runtime/server/plugins/warm-up.js b/dist/runtime/server/plugins/warm-up.js
index 4322aee6566875cd49237fc02a2bff773ba61a3b..a75e07623c1b5688a7f54f0c8c45e69272996b49 100644
--- a/dist/runtime/server/plugins/warm-up.js
+++ b/dist/runtime/server/plugins/warm-up.js
@@ -1,7 +1,7 @@
-import { defineNitroPlugin } from "nitropack/runtime";
+import { definePlugin } from "nitro";
 import { joinURL, withLeadingSlash } from "ufo";
 import { useSitemapRuntimeConfig } from "../utils.js";
-export default defineNitroPlugin((nitroApp) => {
+export default definePlugin((nitroApp) => {
   const { sitemaps, sitemapsPathPrefix } = useSitemapRuntimeConfig();
   const queue = [];
   const timeoutIds = [];
diff --git a/dist/runtime/server/sitemap/builder/sitemap-index.js b/dist/runtime/server/sitemap/builder/sitemap-index.js
index 09fd7f22a1bbc240403c5d6f925235546a50139d..d3be82c6cc2bb3ca7e63c8090d87443fac4fbcdd 100644
--- a/dist/runtime/server/sitemap/builder/sitemap-index.js
+++ b/dist/runtime/server/sitemap/builder/sitemap-index.js
@@ -1,5 +1,5 @@
 import { getHeader } from "h3";
-import { defineCachedFunction } from "nitropack/runtime";
+import { defineCachedFunction } from "nitro/cache";
 import { joinURL, withQuery } from "ufo";
 import staticConfig from "#sitemap-virtual/static-config.mjs";
 import { normaliseDate } from "../urlset/normalise.js";
diff --git a/dist/runtime/server/sitemap/builder/sitemap.js b/dist/runtime/server/sitemap/builder/sitemap.js
index c679de963b94df517c61067b988d83e8d9390d68..ffcc410f9fe88a0d34fe560cab1ea8e5b7224649 100644
--- a/dist/runtime/server/sitemap/builder/sitemap.js
+++ b/dist/runtime/server/sitemap/builder/sitemap.js
@@ -1,5 +1,6 @@
 import { getHeader } from "h3";
-import { defineCachedFunction, useRuntimeConfig } from "nitropack/runtime";
+import { defineCachedFunction } from "nitro/cache";
+import { useRuntimeConfig } from "nitro/runtime-config";
 import { resolveSitePath } from "nuxt-site-config/urls";
 import { joinURL, withHttps } from "ufo";
 import staticConfig from "#sitemap-virtual/static-config.mjs";
diff --git a/dist/runtime/server/sitemap/event-handlers.js b/dist/runtime/server/sitemap/event-handlers.js
index c6834ea3cc72f72ceb056e3b7873c0117fd57ca2..075c1ab0e98965de2466c44bdb5c3cf2bde7e222 100644
--- a/dist/runtime/server/sitemap/event-handlers.js
+++ b/dist/runtime/server/sitemap/event-handlers.js
@@ -1,5 +1,6 @@
 import { appendHeader, createError, getRouterParam, sendRedirect, setHeader } from "h3";
-import { useNitroApp, useRuntimeConfig } from "nitropack/runtime";
+import { useNitroApp } from "nitro/app";
+import { useRuntimeConfig } from "nitro/runtime-config";
 import { joinURL, withBase, withLeadingSlash, withoutLeadingSlash, withoutTrailingSlash } from "ufo";
 import { useSitemapRuntimeConfig } from "../utils.js";
 import { buildSitemapIndex, urlsToIndexXml } from "./builder/sitemap-index.js";
diff --git a/dist/runtime/server/sitemap/nitro.js b/dist/runtime/server/sitemap/nitro.js
index 55e290c6d4c508cbdeb5db8230623c81c00abe42..2ae2401e3e61aa1d5846fe43a9177ca185d3cbaf 100644
--- a/dist/runtime/server/sitemap/nitro.js
+++ b/dist/runtime/server/sitemap/nitro.js
@@ -1,6 +1,7 @@
 import { defu } from "defu";
 import { createError, getHeader, getQuery, setHeader } from "h3";
-import { defineCachedFunction, useNitroApp } from "nitropack/runtime";
+import { useNitroApp } from "nitro/app";
+import { defineCachedFunction } from "nitro/cache";
 import { fixSlashes } from "nuxt-site-config/urls";
 import { getPathRobotConfig } from "#internal/nuxt-robots/getPathRobotConfig";
 import { getSiteConfig } from "#site-config/server/composables/getSiteConfig";
diff --git a/dist/runtime/server/utils.js b/dist/runtime/server/utils.js
index 9e4d8b556dbf948b59996a384099b0731a959bb4..7eb2692ccd04a39e8bbd86fe0e3bdd0add6fed1f 100644
--- a/dist/runtime/server/utils.js
+++ b/dist/runtime/server/utils.js
@@ -1,4 +1,4 @@
-import { useRuntimeConfig } from "nitropack/runtime";
+import { useRuntimeConfig } from "nitro/runtime-config";
 import staticConfig from "#sitemap-virtual/static-config.mjs";
 import { normalizeRuntimeFilters } from "../utils-pure.js";
 export * from "../utils-pure.js";
