diff --git a/dist/runtime/server/plugins/initContext.js b/dist/runtime/server/plugins/initContext.js
--- a/dist/runtime/server/plugins/initContext.js
+++ b/dist/runtime/server/plugins/initContext.js
@@ -29,9 +29,10 @@ export default defineNitroPlugin(async (nitroApp) => {
   }
   if (import.meta.prerender) {
     nitroApp.hooks.hook("render:html", async (ctx, { event }) => {
-      const routeOptions = getRouteRules(event);
+      const routeOptions = event.context.routeRules || getRouteRules(event.req.method, event.url.pathname).routeRules || {};
+      const ssr = routeOptions.ssr?.options;
       const isIsland = process.env.NUXT_COMPONENT_ISLANDS && event.path.startsWith("/__nuxt_island");
-      const noSSR = !!process.env.NUXT_NO_SSR || event.context.nuxt?.noSSR || routeOptions.ssr === false && !isIsland || (import.meta.prerender ? PRERENDER_NO_SSR_ROUTES.has(event.path) : false);
+      const noSSR = !!process.env.NUXT_NO_SSR || event.context.nuxt?.noSSR || ssr === false && !isIsland || (import.meta.prerender ? PRERENDER_NO_SSR_ROUTES.has(event.path) : false);
       if (noSSR) {
         let rule = event.context.robots?.rule;
         if (event.path === "/404.html")
