diff --git a/dist/module.mjs b/dist/module.mjs
index c6401a7..77be442 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -1,6 +1,6 @@
 import fsp, { readFile } from 'node:fs/promises';
 import { useLogger, useNuxt, addDevServerHandler, addVitePlugin, createResolver, extendViteConfig, defineNuxtModule, addTemplate, addBuildPlugin } from '@nuxt/kit';
-import { join, relative, resolve, extname } from 'pathe';
+import { join, relative, resolve, extname, basename } from 'pathe';
 import { joinURL, withTrailingSlash, withLeadingSlash, withoutLeadingSlash } from 'ufo';
 import defu$1, { defu } from 'defu';
 import { transformCSS, resolveMinifyCssEsbuildOptions, normalizeFontData, generateFontFace, parseFont, defaultValues, resolveProviders, createResolver as createResolver$1, defaultOptions } from 'fontless';
@@ -106,8 +106,8 @@ async function setupPublicAssetStrategy(options = {}) {
   };
   nuxt.hook("modules:done", () => nuxt.callHook("fonts:public-asset-context", context));
   async function devEventHandler(event) {
-    const filename = event.path.slice(1);
-    const url = context.renderedFontURLs.get(event.path.slice(1));
+    const filename = basename(event.path);
+    const url = context.renderedFontURLs.get(filename);
     if (!url) {
       throw createError({ statusCode: 404 });
     }
@@ -121,7 +121,7 @@ async function setupPublicAssetStrategy(options = {}) {
     return res;
   }
   addDevServerHandler({
-    route: joinURL(nuxt.options.runtimeConfig.app.baseURL || nuxt.options.app.baseURL, context.assetsBaseURL),
+    route: joinURL(nuxt.options.runtimeConfig.app.baseURL || nuxt.options.app.baseURL, context.assetsBaseURL, "**"),
     handler: eventHandler(devEventHandler)
   });
   addVitePlugin({
@@ -155,7 +155,8 @@ async function setupPublicAssetStrategy(options = {}) {
     publicAssets: [{
       dir: cacheDir,
       maxAge: ONE_YEAR_IN_SECONDS,
-      baseURL: context.assetsBaseURL
+      baseURL: context.assetsBaseURL,
+      fallthrough: nuxt.options.dev
     }],
     ignore: [`!${join(cacheDir, "**/*")}`],
     prerender: {
