diff --git a/lib/esm/Parser.js b/lib/esm/Parser.js
index 68cc0ed4c6a421c4cd64e28a7e9538e6a1b974c8..d652729144ad31be67bf122403aa3e13922b19ad 100644
--- a/lib/esm/Parser.js
+++ b/lib/esm/Parser.js
@@ -1,5 +1,5 @@
 import Tokenizer, { QuoteType } from "./Tokenizer.js";
-import { fromCodePoint } from "entities/lib/decode.js";
+import { fromCodePoint } from "entities/dist/decode.js";
 const formTags = new Set([
     "input",
     "option",
diff --git a/lib/esm/Tokenizer.js b/lib/esm/Tokenizer.js
index a97456b05090e2ff63c6dc5410f6ab2b86f583ef..a179e150bd8bc71bba1826c0dae97feb8116688a 100644
--- a/lib/esm/Tokenizer.js
+++ b/lib/esm/Tokenizer.js
@@ -1,4 +1,4 @@
-import { htmlDecodeTree, xmlDecodeTree, BinTrieFlags, determineBranch, replaceCodePoint, } from "entities/lib/decode.js";
+import { htmlDecodeTree, xmlDecodeTree, BinTrieFlags, determineBranch, replaceCodePoint, } from "entities/dist/decode.js";
 var CharCodes;
 (function (CharCodes) {
     CharCodes[CharCodes["Tab"] = 9] = "Tab";
