diff --git a/dist/runtime/server/composables/utils.js b/dist/runtime/server/composables/utils.js
--- a/dist/runtime/server/composables/utils.js
+++ b/dist/runtime/server/composables/utils.js
@@ -28,7 +28,7 @@ export function withSiteUrl(e, path, options = {}) {
     absolute: true,
     siteUrl,
     trailingSlash: siteConfig.trailingSlash,
-    base: e.context.nitro.baseURL,
+    base: useRuntimeConfig().app.baseURL || "/",
     withBase: options.withBase
   });
 }
diff --git a/dist/runtime/server/middleware/init.js b/dist/runtime/server/middleware/init.js
--- a/dist/runtime/server/middleware/init.js
+++ b/dist/runtime/server/middleware/init.js
@@ -32,10 +32,11 @@ export default eventHandler(async (e) => {
   });
   const buildStack = config.stack || [];
   buildStack.forEach((c) => siteConfig.push(c));
-  if (e.context._nitro.routeRules.site) {
+  const routeSiteConfig = e.context.routeRules?.site?.options;
+  if (routeSiteConfig) {
     siteConfig.push({
       _context: "route-rules",
-      ...e.context._nitro.routeRules.site
+      ...routeSiteConfig
     });
   }
   if (config.multiTenancy) {
diff --git a/dist/runtime/server/plugins/injectState.js b/dist/runtime/server/plugins/injectState.js
--- a/dist/runtime/server/plugins/injectState.js
+++ b/dist/runtime/server/plugins/injectState.js
@@ -1,4 +1,4 @@
-import devalue from "@nuxt/devalue";
+import { uneval as devalue } from "devalue";
 import { defineNitroPlugin, getRouteRules } from "nitropack/runtime";
 import { toValue } from "vue";
 import { NUXT_SITE_CONFIG_NO_SSR } from "#nuxt-site-config/no-ssr.mjs";
@@ -6,10 +6,11 @@ import { getSiteConfig } from "../composables/getSiteConfig.js";
 const PRERENDER_NO_SSR_ROUTES = /* @__PURE__ */ new Set(["/index.html", "/200.html", "/404.html"]);
 export default defineNitroPlugin(async (nitroApp) => {
   nitroApp.hooks.hook("render:html", async (ctx, { event }) => {
-    const routeOptions = getRouteRules(event);
+    const routeOptions = event.context.routeRules || getRouteRules(event.req.method, event.url.pathname).routeRules || {};
+    const ssr = routeOptions.ssr?.options;
     const isIsland = process.env.NUXT_COMPONENT_ISLANDS && event.path.startsWith("/__nuxt_island");
     const url = event.path;
-    const noSSR = !!NUXT_SITE_CONFIG_NO_SSR || !!process.env.NUXT_NO_SSR || event.context.nuxt?.noSSR || routeOptions.ssr === false && !isIsland || (import.meta.prerender ? PRERENDER_NO_SSR_ROUTES.has(url) : false);
+    const noSSR = !!NUXT_SITE_CONFIG_NO_SSR || !!process.env.NUXT_NO_SSR || event.context.nuxt?.noSSR || ssr === false && !isIsland || (import.meta.prerender ? PRERENDER_NO_SSR_ROUTES.has(url) : false);
     if (noSSR) {
       const siteConfig = Object.fromEntries(
         Object.entries(getSiteConfig(event)).map(([k, v]) => [k, toValue(v)])
