diff --git a/dist/runtime/server/routes/__link-checker__/debug.js b/dist/runtime/server/routes/__link-checker__/debug.js
index 560679325d68c2ab5aed6510aab43392551a43e2..e8d08fdb3fcff331903d02a3ed47ab4918571fe5 100644
--- a/dist/runtime/server/routes/__link-checker__/debug.js
+++ b/dist/runtime/server/routes/__link-checker__/debug.js
@@ -1,5 +1,5 @@
 import { defineEventHandler } from "h3";
-import { useRuntimeConfig } from "#imports";
+import { useRuntimeConfig } from "nitro/runtime-config";
 export default defineEventHandler(async (e) => {
   return {
     runtimeConfig: useRuntimeConfig(e).public["nuxt-link-checker"]
diff --git a/dist/runtime/server/routes/__link-checker__/inspect.js b/dist/runtime/server/routes/__link-checker__/inspect.js
index db432f134fcb41fa4eb225085195f201b1489e6a..f7891ccdb52654e2c4f1db56bc3cb055b5f28654 100644
--- a/dist/runtime/server/routes/__link-checker__/inspect.js
+++ b/dist/runtime/server/routes/__link-checker__/inspect.js
@@ -1,9 +1,9 @@
 import { createDefu } from "defu";
 import Fuse from "fuse.js";
 import { createError, defineEventHandler, readBody } from "h3";
+import { useRuntimeConfig } from "nitro/runtime-config";
 import { fixSlashes } from "nuxt-site-config/urls";
 import { resolve } from "pathe";
-import { useRuntimeConfig } from "#imports";
 import { getNitroOrigin } from "#site-config/server/composables/getNitroOrigin";
 import { getSiteConfig } from "#site-config/server/composables/getSiteConfig";
 import { generateFileLinkDiff, generateFileLinkPreviews, getLinkResponse, inspect, isNonFetchableLink, lruFsCache } from "../../../shared/index.js";
diff --git a/dist/runtime/server/routes/__link-checker__/links.js b/dist/runtime/server/routes/__link-checker__/links.js
index 9195da8cc40fac9824d45f651d0642c58c15d802..125c5aa7af14196a6b284584fd988adb637ae01f 100644
--- a/dist/runtime/server/routes/__link-checker__/links.js
+++ b/dist/runtime/server/routes/__link-checker__/links.js
@@ -1,8 +1,8 @@
-import { defineCachedEventHandler } from "nitropack/runtime";
-import { useRuntimeConfig } from "#imports";
+import { defineCachedHandler } from "nitro/cache";
+import { useRuntimeConfig } from "nitro/runtime-config";
 import contentLinkProvider from "#link-checker/content-provider";
 import pagePaths from "#nuxt-link-checker-sitemap/pages.mjs";
-export default defineCachedEventHandler(async (e) => {
+export default defineCachedHandler(async (e) => {
   const runtimeConfig = useRuntimeConfig().public["nuxt-link-checker"] || {};
   const linkDb = [
     ...pagePaths
