diff --git a/dist/index.mjs b/dist/index.mjs
index 4be6902..e60715e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1398,7 +1398,9 @@ function tokenize$3(effects, ok, nok) {
       throw new Error("expected `:`");
     }
     if (self.previous !== null && !markdownLineEndingOrSpace(self.previous) && ![Codes.openingSquareBracket, Codes.star, Codes.underscore, Codes.openingParentheses].includes(self.previous)) {
-      return nok(code);
+      if (!/(?!:)[\p{Script=Han}\p{Script=Hiragana}\p{Script=Katakana}\p{Script=Hangul}\p{Punctuation}]/u.test(String.fromCodePoint(self.previous))) {
+        return nok(code);
+      }
     }
     if (!previous.call(self, self.previous)) {
       throw new Error("expected correct previous");
